src/js/pages/Home/index.js

```diff
@@ -211,6 +215,64 @@ export default class Home extends React.PureComponent {
+                                writers?.filter(e => SOURCE_WRITER_FILTER['D'].includes(e.writerID)).map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))                                
+                              ) : source === 'B' ? (                                 
+                                writers?.filter(e => SOURCE_WRITER_FILTER['B'].includes(e.writerID)).map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))
+                              ) : source === 'A' ? (
+                                writers?.filter(e => !SOURCE_WRITER_FILTER['A'].includes(e.writerID)).map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))                                
+                              ) : source === 'S' ? (                                 
+                                writers?.filter(e => SOURCE_WRITER_FILTER['S'].includes(e.writerID)).map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))
+                              ) : source === 'N' ? (                                 
+                                writers?.filter(e => SOURCE_WRITER_FILTER['N'].includes(e.writerID)).map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))
+                              ) : source === 'R' ? (                                 
+                                writers?.filter(e => SOURCE_WRITER_FILTER['R'].includes(e.writerID)).map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))
+                              ) : (
+                                writers?.map(writer => (
+                                  <option key={writer.writerID} value={writer.writerID}>
+                                    {writer.writerEnglish}
+                                  </option>
+                                ))
+                              )
```

Comment:
I think all this can be instead expressed as:

```js
writers?
  .filter(e => source
    ? SOURCE_WRITER_FILTER[source].includes(e.writerID)
    : true
  )
  .map(writer => (
    <option key={writer.writerID} value={writer.writerID}>
      {writer.writerEnglish}
    </option>
   ))
```

Reply:
Revised, as per need.